tests/components/volvo/test_button.py

```diff
@@ -41,47 +39,58 @@ async def test_button(
     await snapshot_platform(hass, entity_registry, snapshot, mock_config_entry.entry_id)
 
 
-@pytest.mark.usefixtures("mock_api", "full_model")
+@pytest.mark.usefixtures("full_model")
+@pytest.mark.parametrize(
+    "command",
+    ["start_climatization", "stop_climatization", "flash", "honk", "honk_flash"],
+)
 async def test_button_press(
     hass: HomeAssistant,
     setup_integration: Callable[[], Awaitable[bool]],
+    mock_api: VolvoCarsApi,
+    command: str,
 ) -> None:
     """Test button press."""
 
     with patch("homeassistant.components.volvo.PLATFORMS", [Platform.BUTTON]):
         assert await setup_integration()
 
-    assert hass.states.get(_BUTTON_ID).state == STATE_UNKNOWN
+    entity_id = f"button.volvo_xc40_{command}"
 
     await hass.services.async_call(
         BUTTON_DOMAIN,
         SERVICE_PRESS,
-        {ATTR_ENTITY_ID: _BUTTON_ID},
+        {ATTR_ENTITY_ID: entity_id},
         blocking=True,
     )
     await hass.async_block_till_done()
-    assert hass.states.get(_BUTTON_ID).state != STATE_UNKNOWN
 
+    assert len(mock_api.async_execute_command.mock_calls) == 1
 
-@pytest.mark.usefixtures("mock_api", "full_model")
+
+@pytest.mark.usefixtures("full_model")
+@pytest.mark.parametrize(
+    "command",
+    ["start_climatization", "stop_climatization", "flash", "honk", "honk_flash"],
+)
 async def test_button_press_error(
     hass: HomeAssistant,
     setup_integration: Callable[[], Awaitable[bool]],
     mock_api: VolvoCarsApi,
+    command: str,
 ) -> None:
     """Test button press with error response."""
 
     with patch("homeassistant.components.volvo.PLATFORMS", [Platform.BUTTON]):
         assert await setup_integration()
 
-    assert hass.states.get(_BUTTON_ID).state == STATE_UNKNOWN
-
+    entity_id = f"button.volvo_xc40_{command}"
     configure_mock(mock_api.async_execute_command, side_effect=VolvoApiException)
 
     with pytest.raises(HomeAssistantError):
         await hass.services.async_call(
             BUTTON_DOMAIN,
             SERVICE_PRESS,
-            {ATTR_ENTITY_ID: _BUTTON_ID},
+            {ATTR_ENTITY_ID: entity_id},
```

Comment:
```suggestion
configure_mock(mock_api.async_execute_command, side_effect=VolvoApiException)

with pytest.raises(HomeAssistantError):
await hass.services.async_call(
BUTTON_DOMAIN,
SERVICE_PRESS,
            {ATTR_ENTITY_ID: f"button.volvo_xc40_{command}"},
```
It has a single usage right? Then it can be put directly in the `ATTR_ENTITY_ID`.